src/zenml/zen_stores/schemas/run_metadata_schemas.py

```diff
@@ -103,3 +78,63 @@ class RunMetadataSchema(BaseSchema, table=True):
     key: str
     value: str = Field(sa_column=Column(TEXT, nullable=False))
     type: str
+
+    publisher_step_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=StepRunSchema.__tablename__,
+        source_column="publisher_step_id",
+        target_column="id",
+        ondelete="SET NULL",
+        nullable=True,
+    )
+
+
+class RunMetadataResourceSchema(SQLModel, table=True):
+    """Table for linking resources to run metadata entries."""
+
+    __tablename__ = "run_metadata_resource"
+
+    id: UUID = Field(default_factory=uuid4, primary_key=True)
+    resource_id: UUID
+    resource_type: str = Field(sa_column=Column(VARCHAR(255), nullable=False))
+    run_metadata_id: UUID = build_foreign_key_field(
+        source=__tablename__,
+        target=RunMetadataSchema.__tablename__,
+        source_column="run_metadata_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=False,
+    )
+
+    # Relationship back to the base metadata table
+    run_metadata: RunMetadataSchema = Relationship(back_populates="resources")
+
+    # Relationship to link specific resource types
+    pipeline_run: List["PipelineRunSchema"] = Relationship(
```

Comment:
```suggestion
    pipeline_runs: List["PipelineRunSchema"] = Relationship(
```

Same for all the others